src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -88,15 +91,21 @@
   // -1 would mean forever until one side is eliminated (the default is infinite)
   private final int m_maxRounds;
 
+  /**
+   *  Constructor. Suppress checkstyle warning.
+   */
   public MustFightBattle(final Territory battleSite, final PlayerID attacker, final GameData data,
       final BattleTracker battleTracker) {
-    super(battleSite, attacker, battleTracker, false, BattleType.NORMAL, data);
+    super(battleSite, attacker, battleTracker, data);
     m_defendingUnits.addAll(m_battleSite.getUnits().getMatches(Matches.enemyUnit(attacker, data)));
     if (battleSite.isWater()) {
       m_maxRounds = games.strategy.triplea.Properties.getSeaBattleRounds(data);
     } else {
       m_maxRounds = games.strategy.triplea.Properties.getLandBattleRounds(data);
     }
+    m_attackingFromMap = new HashMap<>();
+    m_attackingFrom = new HashSet<>();
+    m_amphibiousAttackFrom = new ArrayList<>();
```

Comment:
It looks like these fields are already initialized when they are declared above.  Is it necessary to do it again in the constructor?

Also, IIRC, `MustFightBattle#m_amphibiousAttackFrom` was `final` prior to #1583.  Can it be made so again?